torchmetrics/classification/calibration_error.py

```diff
@@ -0,0 +1,112 @@
+from typing import Any, Callable, Optional
+
+import torch
+from torch import Tensor
+
+from torchmetrics.functional.classification.calibration_error import _ce_compute, _ce_update
+from torchmetrics.metric import Metric
+from torchmetrics.utilities import rank_zero_warn
+
+
+class CalibrationError(Metric):
+
+    def __init__(
+        self,
+        n_bins: int = 15,
+        norm: str = "l1",
+        compute_on_step: bool = False,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None
+    ):
+        """
+
+        Computes the top-label calibration error as described in `https://arxiv.org/pdf/1909.10155.pdf`.
+
+        Three different norms are implemented, each corresponding to variations on the calibration error metric.
+
+        L1 norm (Expected Calibration Error)
+
+        .. math::
+            \text{Accuracy} = \frac{1}{N}\sum_i^N \|(p_i - c_i)\|
+
+        Infinity norm (Maximum Calibration Error)
+
+        .. math::
+        \text{Accuracy} =  \max_{i} (p_i - c_i)
+
+        L2 norm (Root Mean Square Calibration Error)
+
+        .. math::
+        \text{Accuracy} = \frac{1}{N}\sum_i^N (p_i - c_i)^2
+
+        Where p_i is the top-1 prediction accuracy in bin i and c_i is the average confidence of predictions in bin i.
+
+        # NOTE: L2-norm debiasing is not yet supported.
+
+
+        Args:
+            n_bins (int, optional): Number of bins to use when computing t. Defaults to 15.
+            norm (str, optional): Norm used to compare empirical and expected probability bins.
+                Defaults to "l1", or Expected Calibration Error.
+            debias (bool, optional): Applies debiasing term, only implemented for l2 norm. Defaults to True.
+            compute_on_step (bool, optional):  Forward only calls ``update()`` and return None if this is set to False. Defaults to False.
+            dist_sync_on_step (bool, optional): Synchronize metric state across processes at each ``forward()``
+                before returning the value at the step.. Defaults to False.
+            process_group (Optional[Any], optional): Specify the process group on which synchronization is called. default: None (which selects the entire world). Defaults to None.
+            dist_sync_fn (Callable, optional): Callback that performs the ``allgather`` operation on the metric state. When ``None``, DDP
+                will be used to perform the ``allgather``.. Defaults to None.
+        """
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn
+        )
+
+        if norm not in ["l1", "l2", "max"]:
+            raise ValueError(f"Norm {norm} is not supported. Please select from l1, l2, or max. ")
+
+        self.n_bins = n_bins
+        self.bin_boundaries = torch.linspace(0, 1, n_bins + 1)
```

Comment:
this will make sure the bins get moved to the correct device if the metric is moved
```suggestion
        self.register_buffer("bin_boundaries, torch.linspace(0, 1, n_bins + 1))
```